spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -387,6 +387,28 @@ will run in the context of an LRA and controls whether or not:
 - to throw an exception if the method returns particular HTTP status codes.
 - to enlist the class as a participant of the LRA.
 
+These operations are performed before the execution proceeds to the invoked
+`@LRA` annotated business method. If any of the LRA operations prohibits the
+execution of the business method, the LRA implemenation MUST return one of the
+following status codes to user invoking the LRA annotated method. This means
+that in these conditions the business method IS NOT invoked.
+
+|===
+|Status code |Conditions
+
+| 410
+a| - the passed LRA context has already ended
+   - the passed LRA context is not found
+
+| 412
+a| - the user passed the LRA context to LRA type which prohibits it (see LRA.Type)
+   - the content of the LRA context is invalid
+
+| 5xx
+a| - other unspecified LRA processing errors
```

Comment:
Maybe this could be articulated better but the `5xx` means an unspecified reason, doesn't it? It's something the spec does not define. We consider there could be thrown e.g. a `RuntimeException` but it's just one of possible reasons.
I think other spec uses the similar wording to express that some by spec unknown error happens (https://docs.oracle.com/javaee/7/api/javax/transaction/SystemException.html).
The wording is quite fine for me.

Reply:
thanks @ochaloup. @mmusgrov WDYT about Ondras response?

Reply:
If we specify that 5xx has an unspecified meaning for the @LRA annotation then it would preclude using any of those codes to mean something specific in a later spec revision.

[Note that we already give the 500 code a specific semantic interpretation for participant responses to the @complete and @compensate calls - it means the response body does not correspond to a valid participant status].

Reply:
At most I would say *500 Internal Server Error*. The other 5xx codes mean something specific.

Reply:
We can still change the status codes in later revisions, we are allowed to make breaking changes. But I accept your suggestion and I will change it to  500 Internal Server Error.

Reply:
We can make breaking changes but we shouldn't unless we have a good reason to.

I still do not see the benefit of specifying it though since leaving it unspecified would give us the maximum latitude to reserve the code for any future need.

Reply:
ok, I will remove it. It's not that important for me. I just wanted to show that even LRA code can fail with unexpected errors and in this case the execution also does not proceed to the users business method.

Reply:
The implementation should not be throwing unspecified errors. But the infrastructure that the implementation is running on may, however it is not our responsibility to define what those are.